ProfileManager.js

```diff
@@ -136,14 +145,17 @@ export default class ProfileManager {
     const capabilityKey = `${capabilityCacheKey}-${id}-${useEphemeralSigner}`;
 
     const capabilityCache = this._getCache(capabilityCacheKey);
-    const agentZcap = capabilityCache.get(capabilityKey);
+    let agentZcap = capabilityCache.get(capabilityKey);
     if(agentZcap) {
       return agentZcap;
     }
-
     const promise = this._getAgentCapability(
       {id, profileAgent, useEphemeralSigner});
-    capabilityCache.set(capabilityKey, promise);
+    agentZcap = await promise;
+    const now = Date.now();
+    const expiryDate = new Date(agentZcap.expires);
```

Comment:
I think we need to do something like this in case the zcap has no `expires`:

```suggestion
    const expiryDate = new Date(agentZcap.expires || (now + this.zcapTtl));
```